components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -172,6 +172,35 @@ public static IdentityProvider buildFederatedIdentityProvider(HttpServletRequest
                         customAuthenticatorNames.add(new String(value, StandardCharsets.UTF_8));
                     } else if (key.startsWith("custom_pro_name")) {
                         proConnectorNames.add(new String(value, StandardCharsets.UTF_8));
+                    } else if (key.endsWith("_propEnabled")) {
```

Comment:
aren't the below if conditions also share common logic? may be move to a method

Reply:
Some lines seems similar, still there are major differences of the value types and the substring indices. Moving to another method might add bit complexity with only 2 usages for now. WDYT ?